template/golang-http/Dockerfile

```diff
@@ -8,41 +8,54 @@ ARG TARGETARCH
 
 RUN apk --no-cache add git
 
-ENV CGO_ENABLED=0
-
 COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
 RUN chmod +x /usr/bin/fwatchdog
 
+ENV CGO_ENABLED=0
+
 RUN mkdir -p /go/src/handler
 WORKDIR /go/src/handler
 COPY . .
 
+ARG GO111MODULE="off"
+ARG GOPROXY=""
+ARG GOFLAGS=""
+
+# Add user overrides to the root go.mod, which is the only place "replace" can be used
+RUN cat function/GO_REPLACE.txt >> ./go.mod || exit 0
+RUN if [ -d ./function/vendor ]; then \
+    echo "moving handler vendor" && \
+    mv -f ./function/vendor . && go mod tidy; \
```

Comment:
I expected to see the mv, but what does the `go mod tidy` part do?

Reply:
it ensures that the modules.txt or the go.sum file is up-to-date, there is just enough information for it to fail some consistency checks even though we know it will be fine

Reply:
I did some more testing i don't think this works as expected so I am removing it